apps/challenges/github_utils.py

```diff
@@ -0,0 +1,213 @@
+
+    def get_data_from_path(self, path):
+        """
+        Gets the file data in string format in the repository at particular path
+        Calls get_content_from_path and encode the base64 content
+        """
+        content_response = self.get_content_from_path(path)
+        string_data = None
+        if content_response and content_response.get("content"):
+            string_data = base64.b64decode(content_response["content"]).decode(
+                "utf-8"
+            )
+        return string_data
+
+    def update_content_from_path(self, path, content):
+        """
+        Updates the file content, creates a commit in the repository at particular path
+        Ref: https://docs.github.com/en/rest/reference/repos#create-or-update-file-contents
+        """
+        url = URLS.get("contents").format(self.GITHUB_REPOSITORY, path)
+        data = {
+            "message": self.COMMIT_PREFIX.format(path),
+            "branch": self.BRANCH,
+            "sha": self.get_content_from_path(path).get("sha"),
+            "content": content,
+        }
+        response = self.make_request(url, "PUT", data=data)
+        return response
+
+    def update_data_from_path(self, path, data):
+        """
+        Updates the file data to the data(string) provided, at particular path
+        Call update_content_from_path with decoded base64 content
+        """
+        content = base64.b64encode(bytes(data, "utf-8")).decode("utf-8")
+        return self.update_content_from_path(path, content)
+
+    def is_repository(self):
+        url = URLS.get("repos").format(self.GITHUB_REPOSITORY)
+        repo_response = self.make_request(url, "GET")
+        return True if repo_response else False
+
+
+@app.task
+def github_challenge_sync(challenge):
+    challenge = deserialize_object(challenge)
+    github = GithubInterface(
+        GITHUB_REPOSITORY=getattr(challenge, "github_repository"),
+        GITHUB_AUTH_TOKEN=getattr(challenge, "github_token"),
+    )
+    if not github.is_repository():
+        return
+    try:
+        # Challenge Non-file field Update
+        challenge_config_str = github.get_data_from_path(
+            "challenge_config.yaml"
+        )
+        challenge_config_yaml = yaml.safe_load(challenge_config_str)
+        update_challenge_config = False
+        for field in challenge_non_file_fields:
+            # Ignoring commits when no update in field value
+            if challenge_config_yaml.get(
+                field
+            ) is not None and challenge_config_yaml[field] == getattr(
+                challenge, field
+            ):
+                continue
+            update_challenge_config = True
+            challenge_config_yaml[field] = getattr(challenge, field)
+        if update_challenge_config:
+            content_str = yaml.dump(challenge_config_yaml, sort_keys=False)
+            github.update_data_from_path("challenge_config.yaml", content_str)
+
+        # Challenge File fields Update
```

Comment:
Change to `Challenge file fields update`

Reply:
Updated.